ddl/pause_test.go

```diff
@@ -0,0 +1,306 @@
+	// Drop schema.
+	{"drop database test_create_db", true, model.StatePublic, true, false, nil},
+	{"drop database test_create_db", false, model.StateWriteOnly, true, true, []string{"create database if not exists test_create_db"}},
+	{"drop database test_create_db", false, model.StateDeleteOnly, true, true, []string{"create database if not exists test_create_db"}},
+	{"drop database test_create_db", false, model.StateNone, false, true, []string{"create database if not exists test_create_db"}},
+
+	// Drop column.
+	{"alter table t_user drop column c3", true, model.StatePublic, true, false, nil},
+	{"alter table t_user drop column c3", false, model.StateDeleteOnly, true, false, nil},
+	{"alter table t_user drop column c3", false, model.StateDeleteOnly, false, true, []string{"alter table t_user add column c3 bigint"}},
+	{"alter table t_user drop column c3", false, model.StateWriteOnly, true, true, []string{"alter table t_user add column c3 bigint"}},
+	{"alter table t_user drop column c3", false, model.StateDeleteReorganization, true, true, []string{"alter table t_user add column c3 bigint"}},
+	{"alter table t_user drop column c3", false, model.StateNone, false, true, []string{"alter table t_user add column c3 bigint"}},
+
+	// Drop column with index.
+	{"alter table t_user drop column c3", true, model.StatePublic, true, false, []string{"alter table t_user add column c3 bigint", "alter table t_user add index idx_c3(c3)"}},
+	{"alter table t_user drop column c3", false, model.StateDeleteOnly, true, false, nil},
+	{"alter table t_user drop column c3", false, model.StateDeleteOnly, false, true, []string{"alter table t_user add column c3 bigint", "alter table t_user add index idx_c3(c3)"}},
+	{"alter table t_user drop column c3", false, model.StateWriteOnly, true, true, []string{"alter table t_user add column c3 bigint", "alter table t_user add index idx_c3(c3)"}},
+	{"alter table t_user drop column c3", false, model.StateDeleteReorganization, true, true, []string{"alter table t_user add column c3 bigint", "alter table t_user add index idx_c3(c3)"}},
+	{"alter table t_user drop column c3", false, model.StateNone, false, true, []string{"alter table t_user add column c3 bigint", "alter table t_user add index idx_c3(c3)"}},
+
+	// Modify column, no reorg.
+	{"alter table t_user modify column c11 mediumint", true, model.StateNone, true, false, nil},
+	{"alter table t_user modify column c11 int", false, model.StatePublic, false, true, nil},
+
+	// Modify column, reorg.
+	{"alter table t_user modify column c11 char(10)", true, model.StateNone, true, false, nil},
+	{"alter table t_user modify column c11 char(10)", true, model.StateDeleteOnly, true, true, nil},
+	{"alter table t_user modify column c11 char(10)", true, model.StateWriteOnly, true, true, nil},
+	{"alter table t_user modify column c11 char(10)", true, model.StateWriteReorganization, true, true, nil},
+	{"alter table t_user modify column c11 char(10)", false, model.StatePublic, false, true, nil},
```

Comment:
How about other test cases like `exchange partition`, `add foreign key`, `temporary table`, and so on?

Reply:
Rest LGTM

Reply:
`exchange partition`, `foreign key` and `temporary table` are barely used, looks like not that necessary. It could be added.

Reply:
And also, these cases only need to verify model.StateNone and model.StatePublic, which has been checked in other cases, so that it looks like not necessary.

Reply:
OK. So how about "rename table"?

Reply:
There is no `data change` or `reorganization` during the rename, and the model.JobStateDone is definitely very short, it is not necessary to add such cases.